### PR TITLE
Fixes propagation error in options when setting generic values

### PR DESCRIPTION
**Pull request by JGSweets**
Previously, `is_enabled` would propagate to all `is_enabled` values which may not be intuitive when calling `options.set(dict(is_enabled=True))`.

Now, if the user wants it to propagate to all properties or a specific set they must either use a wildcard or specify the set of options e.g.
```python
options.set({"*.is_enabled": True})
# or
options.set({"int.is_enabled": True})
```

Also resolves issues with tests which used set and erroneously propagated checks.

### CODE REVIEW

File: dataprofiler/tests/profilers/profiler_options/test_boolean_option.py [hunk 1]
```diff
@@ -59,8 +59,6 @@ def test_validate_helper(self):
         # Option is_enabled is not a boolean
         option = self.get_options(is_enabled="Hello World")
         expected_error = ["{}.is_enabled must be a Boolean.".format(optpth)]
-        expected_error += ["{}.{}.is_enabled must be a Boolean." 
-                           .format(optpth, key) for key in self.keys]
```
Comment (ChrisWallace2020) [inline dataprofiler/tests/profilers/profiler_options/test_boolean_option.py:63]: Why did you delete this?

File: dataprofiler/tests/profilers/profiler_options/test_profiler_options.py [hunk 2]
```diff
@@ -43,10 +43,25 @@ def test_default_profiler_options(self, *mocks):
         # Stored in Profiler as StructuredOptions
         self.assertEqual(profile2.options, options.structured_options)
 
+    def test_set_failures(self, *mocks):
+
+        options = ProfilerOptions()
+
+        # check if no '*.' it raises an error bc attribute not found
+        expected_error = "type object '' has no attribute 'is_enabled'"
```
[the remark this answers is not in the document]
Comment (JGSweets) [reply dataprofiler/tests/profilers/profiler_options/test_profiler_options.py:51]: done

File: dataprofiler/tests/profilers/profiler_options/test_boolean_option.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/tests/profilers/profiler_options/test_boolean_option.py:63]: bc is_enabled no longer propagates setting everything at every level unless it is specified as "*.is_enabled"

Comment (ChrisWallace2020) [reply dataprofiler/tests/profilers/profiler_options/test_boolean_option.py:63]: Ah I see

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by gme5078 — APPROVED** [review]
